core/src/test/java/org/opensearch/sql/expression/datetime/DateTimeFunctionTest.java

```diff
@@ -1799,6 +1806,60 @@ public void to_days() {
     assertEquals(longValue(719527L), eval(expression));
   }
 
+  private static Stream<Arguments> getTestDataForToSeconds() {
+    return Stream.of(
+        Arguments.of(new ExprLongValue(950501), new ExprLongValue(62966505600L)),
+        Arguments.of(new ExprStringValue("2009-11-29"), new ExprLongValue(63426672000L)),
+        Arguments.of(new ExprStringValue("2009-11-29 13:43:32"), new ExprLongValue(63426721412L)),
+        Arguments.of(new ExprDateValue("2009-11-29"), new ExprLongValue(63426672000L)),
+        Arguments.of(new ExprDatetimeValue("2009-11-29 13:43:32"), new ExprLongValue(63426721412L)),
+        Arguments.of(new ExprTimestampValue("2009-11-29 13:43:32"), new ExprLongValue(63426721412L)),
+        Arguments.of(new ExprStringValue("0000-00-00"), ExprNullValue.of()),
+        Arguments.of(new ExprStringValue("0000-01-01"), new ExprLongValue(86400L))
+    );
+  }
+
+  @ParameterizedTest
+  @MethodSource("getTestDataForToSeconds")
+  public void testToSeconds(ExprValue arg, ExprValue expected) {
+    lenient().when(nullRef.valueOf(env)).thenReturn(nullValue());
+    lenient().when(missingRef.valueOf(env)).thenReturn(missingValue());
+
+    FunctionExpression expr = DSL.to_seconds(DSL.literal(arg));
+    assertEquals(LONG, expr.type());
+    assertEquals(expected, eval(expr));
+  }
+
+  @Test
+  public void testToSecondsWithTimeType() {
+    lenient().when(nullRef.valueOf(env)).thenReturn(nullValue());
+    lenient().when(missingRef.valueOf(env)).thenReturn(missingValue());
+
+    FunctionExpression expr = DSL.to_seconds(functionProperties, DSL.literal(new ExprTimeValue("10:11:12")));
+
+    long expected = SECONDS_FROM_0001_01_01_TO_EPOCH_START +
+        LocalDate.now(functionProperties.getQueryStartClock())
+            .toEpochSecond(LocalTime.parse("10:11:12"), ZoneOffset.UTC);
+
+    assertEquals(expected, eval(expr).longValue());
+  }
+
+  private static Stream<Arguments> getInvalidTestDataForToSeconds() {
+    return Stream.of(
+        Arguments.of(new ExprLongValue(-123L))
+    );
+  }
+
+  @ParameterizedTest
+  @MethodSource("getInvalidTestDataForToSeconds")
+  public void testToSecondsInvalidArg(ExprValue arg) {
+    lenient().when(nullRef.valueOf(env)).thenReturn(nullValue());
+    lenient().when(missingRef.valueOf(env)).thenReturn(missingValue());
+
+    FunctionExpression expr = DSL.to_seconds(DSL.literal(arg));
+    assertThrows(DateTimeException.class, () -> eval(expr));
+  }
+
```

Comment:
add a test
```
to_seconds(date1) + to_seconds(date2) = to_seconds(date_add(date1, date2))
```

Reply:
I don't think this can be made into a test. The `date_add` function does not add 2 dates together. It adds an interval to the given date (e.g., you can add 31 Days to a given date argument). None of the other "adding" functions for datetime seem to be able to add dates either.

Examples:
[adddate](https://github.com/opensearch-project/sql/blob/main/docs/user/dql/functions.rst#adddate)
[date_add](https://github.com/opensearch-project/sql/blob/main/docs/user/dql/functions.rst#date-add)
[period_add](https://github.com/opensearch-project/sql/blob/main/docs/user/dql/functions.rst#period-add)

Reply:
Oh, right. Is it possible to do something like this?
```
to_seconds(date) + interval.toSeconds() = to_seconds(date_add(date, interval))
`

Reply:
Added in https://github.com/Bit-Quill/opensearch-project-sql/pull/232/commits/fb3c19b88ede07ecc2a25c97cb3efe8cbbf8b730